libbeat/features/features.go

```diff
@@ -88,30 +92,34 @@ func UpdateFromConfig(c *conf.C) error {
 }
 
 func (f *fflags) SetFQDNEnabled(newValue bool) {
-	f.mu.Lock()
-	defer f.mu.Unlock()
+	f.callbackMut.Lock()
+	defer f.callbackMut.Unlock()
 
+	f.stateMut.Lock()
```

Comment:
probably better is for `fqdnEnabled` to be an `atomic.Bool`, then here you can call `oldValue := f.fqdnEnabled.Swap(newValue)` without needing a mutex